import-export-cli/integration/testUtils.go

```diff
@@ -548,7 +645,91 @@ func validateAPIsEqual(t *testing.T, api1 *apim.API, api2 *apim.API) {
 	apim.SortAPIMembers(&api2Copy)
 
 	assert.Equal(t, api1Copy, api2Copy, "API obejcts are not equal")
+}
+
+func validateAPIsList(t *testing.T, args *apiImportExportTestArgs) {
+	t.Helper()
+
+	// Setup apictl envs
+	base.SetupEnv(t, args.srcAPIM.GetEnvName(), args.srcAPIM.GetApimURL(), args.srcAPIM.GetTokenURL())
+
+	// List APIs of env 1
+	base.Login(t, args.srcAPIM.GetEnvName(), args.ctlUser.username, args.ctlUser.password)
+
+	output, _ := listAPIs(t, args)
+
+	apisList := args.srcAPIM.GetAPIs()
+
+	validateListAPIsEqual(t, output, apisList)
+}
+
+func validateListAPIsEqual(t *testing.T, apisListFromCtl string, apisList *apim.APIList) {
+
+	for _, api := range apisList.List {
+		// If the output string contains the same API ID, then decrement the count
+		if strings.Contains(apisListFromCtl, api.ID) {
+			apisList.Count = apisList.Count - 1
+		}
+	}
+
+	// Count == 0 means that all the APIs from apisList were in apisListFromCtl
+	assert.Equal(t, apisList.Count, 0, "API lists are not equal")
+}
+
+func validateAPIsEqualCrossTenant(t *testing.T, api1 *apim.API, api2 *apim.API) {
+	t.Helper()
+
+	api1Copy := apim.CopyAPI(api1)
+	api2Copy := apim.CopyAPI(api2)
+
+	same := "override_with_same_value"
+	// Since the APIs are from too different envs, their respective ID will defer.
+	// Therefore this will be overriden to the same value to ensure that the equality check will pass.
+	api1Copy.ID = same
+	api2Copy.ID = same
+
+	api1Copy.CreatedTime = same
+	api2Copy.CreatedTime = same
+
+	api1Copy.LastUpdatedTime = same
+	api2Copy.LastUpdatedTime = same
+
+	// The contexts and providers will differ since this is a cross tenant import
+	// Therefore this will be overriden to the same value to ensure that the equality check will pass.
+	api1Copy.Context = same
+	api2Copy.Context = same
+
+	api1Copy.Provider = same
+	api2Copy.Provider = same
+
+	// Sort member collections to make equality chack possible
+	apim.SortAPIMembers(&api1Copy)
+	apim.SortAPIMembers(&api2Copy)
+
+	assert.Equal(t, api1Copy, api2Copy, "API obejcts are not equal")
+}
+
+func validateAPIDelete(t *testing.T, args *apiImportExportTestArgs) {
+	t.Helper()
+
+	// Setup apictl envs
+	base.SetupEnv(t, args.srcAPIM.GetEnvName(), args.srcAPIM.GetApimURL(), args.srcAPIM.GetTokenURL())
+
+	// Delete an API of env 1
+	base.Login(t, args.srcAPIM.GetEnvName(), args.ctlUser.username, args.ctlUser.password)
+
+	time.Sleep(1 * time.Second)
+	apisListBeforeDelete := args.srcAPIM.GetAPIs()
+
+	deleteAPIByCtl(t, args)
+
+	apisListAfterDelete := args.srcAPIM.GetAPIs()
+	time.Sleep(1 * time.Second)
+
+	fmt.Println(apisListBeforeDelete.Count)
+	fmt.Println(apisListAfterDelete.Count)
 
+	assert.Equal(t, apisListBeforeDelete.Count, apisListAfterDelete.Count+1, "API delete is not successful")
```

Comment:
Need to loop `apisListAfterDelete` and assert that deleted API does not exist among the remaining APIs.

Reply:
Sure